tool/src/cpp2/header.rs

```diff
@@ -80,11 +70,39 @@ impl Header {
             path,
             includes: BTreeSet::new(),
             decl_include: None,
-            forwards: BTreeSet::new(),
+            forwards: BTreeMap::new(),
             body: String::new(),
             indent_str: "  ",
         }
     }
+
+    fn forward_for(def: TypeDef, ty_name_unnamespaced: &str) -> Forward {
+        match def {
+            TypeDef::Enum(..) => Forward::EnumStruct(ty_name_unnamespaced.into()),
+            TypeDef::Opaque(..) => Forward::Class(ty_name_unnamespaced.into()),
+            TypeDef::Struct(..) | TypeDef::OutStruct(..) => {
+                Forward::Struct(ty_name_unnamespaced.into())
+            }
+            _ => unimplemented!("no other TypeDef variants!"),
+        }
+    }
+
+    pub fn append_forward(&mut self, def: TypeDef, ty_name_unnamespaced: &str) {
+        let forward = Self::forward_for(def, ty_name_unnamespaced);
+
+        let ns = def.attrs().namespace.clone();
```

Comment:
Suggestion: We should make a type `Namespace`. It shouldn't be a string, because:

1. It could have multiple levels which some backends may care about
2. There are so many types flying around in Diplomat that I have tried hard to use newtypes whenever possible to prevent coding errors like passing the wrong string as a positional argument

Reply:
https://github.com/rust-diplomat/diplomat/issues/new